pymodm/base/options.py

```diff
@@ -75,23 +77,41 @@ def auto_dereference(self, auto_dereference):
         self._auto_dereference = auto_dereference
 
     def get_field(self, field_name):
-        """Retrieve a Field instance with the given name."""
+        """Retrieve a Field instance with the given MongoDB name."""
         return self.fields_dict.get(field_name)
 
+    def get_field_from_attname(self, attname):
+        """Retrieve a Fields instance with the given attribute name."""
+        return self.fields_attname_dict.get(attname)
+
+    def _add_field(self, field_inst):
+        """Helper for add_field."""
+        self.fields_dict[field_inst.mongo_name] = field_inst
+        self.fields_attname_dict[field_inst.attname] = field_inst
+        index = bisect(self.fields_ordered, field_inst)
+        self.fields_ordered.insert(index, field_inst)
+
     def add_field(self, field_inst):
         """Add or replace a Field with a given name."""
         field_name = field_inst.mongo_name
```

Comment:
Nit: Since we're keeping track of two names now, it would be helpful to rename `field_name` to something more descriptive like `mongo_name`, so that it's clear which one we're talking about.